agents/agents-mcp/src/jvmTest/kotlin/ai/koog/agents/mcp/DefaultMcpToolDescriptorParserTest.kt

```diff
@@ -195,6 +196,116 @@ class DefaultMcpToolDescriptorParserTest {
+        expectedOptions.forEachIndexed { index, option ->
+            assertEquals(option, enumType.entries[index])
+        }
+    }
+
+    @Test
+    fun `test parsing object parameter with additional properties`() {
+        // Create an SDK Tool with an object parameter that has additional properties
+        val sdkTool = createSdkTool(
+            name = "test-tool",
+            description = "A test tool with object parameter",
+            properties = buildJsonObject {
+                putJsonObject("objectParam") {
+                    put("type", "object")
+                    put("description", "Object parameter")
+                    putJsonObject("properties") {
+                        putJsonObject("name") {
+                            put("type", "string")
+                            put("description", "Name property")
+                        }
+                        putJsonObject("age") {
+                            put("type", "integer")
+                            put("description", "Age property")
+                        }
+                    }
+                    putJsonArray("required") {
+                        add("name")
+                    }
+                    putJsonObject("additionalProperties") {
+                        put("type", "string")
+                    }
+                }
+            },
+            required = listOf("objectParam")
+        )
+
+        // Parse the tool
+        val toolDescriptor = parser.parse(sdkTool)
+
+        // Verify the result
+        val expectedToolDescriptor = ToolDescriptor(
+            name = "test-tool",
+            description = "A test tool with object parameter",
+            requiredParameters = listOf(
+                ToolParameterDescriptor(
+                    name = "objectParam",
+                    description = "Object parameter",
+                    type = ToolParameterType.Object(
+                        properties = listOf(
+                            ToolParameterDescriptor(
+                                name = "name",
+                                description = "Object parameter",
+                                type = ToolParameterType.String
+                            ),
+                            ToolParameterDescriptor(
+                                name = "age",
+                                description = "Object parameter",
+                                type = ToolParameterType.Integer
+                            )
+                        ),
+                        requiredProperties = listOf("name"),
+                        additionalPropertiesType = ToolParameterType.String,
+                        additionalProperties = true
+                    )
+                )
+            ),
+            optionalParameters = emptyList()
+        )
+        assertEquals(expectedToolDescriptor, toolDescriptor)
+    }
+
     @Test
     fun `test error cases`() {
```

Comment:
Could you please split this test and extract each case into a separate test? It will allow to easily identify problems with parsing when appears.